src/ledger/LedgerTxn.cpp

```diff
@@ -26,7 +26,89 @@
 
 namespace stellar
 {
-
+namespace
+{
+void
+enforcePreLoadInvariants(LedgerKey const& key, uint32_t ledgerVersion)
+{
+    switch (key.type())
+    {
+    case ACCOUNT:
+        break;
+    case DATA:
+        break;
+    case OFFER:
+        // Ensure offer is not XLM -- possibly?
+        // special case for offer loading from allow trust op frame?
+        // See LedgerTxnOfferSQL.cpp::loadOffersByAccountAndAsset
```

Comment:
`loadOffersByAccountAndAsset` is actually only ever performed on SQL. Even if `BucketListDB` is enabled, we still keep the offers table around for this very query. It's very expensive to run on a DB without relational tables. You should be fine omitting the check here, but it might be good to include a comment as to why we don't need that specific invariant for BucketListDB.

Reply:
Ok, that makes sense, thanks!